llvm/lib/Target/AIE/AIEMachineScheduler.cpp

```diff
@@ -439,6 +468,13 @@ SUnit *AIEPostRASchedStrategy::pickNodeAndCycle(
     // RegionBottomUpCycles.
     LLVM_DEBUG(dbgs() << "*** Switching to top-down ***\n");
     IsTopDown = true;
+  } else if (IsTopDown && RegionTopDownCycles &&
+             (Top.getCurrCycle() >= RegionTopDownCycles ||
+              mustSwitchToBottomUp())) {
```

Comment:
Checking: I think we will switch back to bottom-up if:
 - We are about to pick an instruction to schedule in a cycle beyond the "top-fixed" cycles.
 - Or we are still within the "top-fixed" cycles", but no instruction can be scheduled top-down anymore. I believe this happens when the remaining instructions are either "bottom-fixed", or have delay slots.